frame/support/src/storage/bounded_vec.rs

```diff
@@ -188,11 +201,26 @@ impl<T: BoundedVecValue, S: Get<u32>> sp_std::ops::Deref for BoundedVec<T, S> {
 	}
 }
 
+impl<T: BoundedVecValue, S: Get<u32>> sp_std::ops::DerefMut for BoundedVec<T, S> {
```

Comment:
This allows you to call methods like `extend` and `push` without checking, not a good idea. 

overall, bounded vec should only coerce to vec when methods are **safe regarding the bound** (cannot expand and re allocate). This is why I only did deref initially. 

Same comments apply to indexMut and AsRefMut: if you can expand the size with them, we can't add them and instead it should be a checked method on the type itself, or just manually convert to vec and try and convert it back.

Reply:
I'm going with DerefMut, this shouldn't be implelemented as said by @kianenigma

However, IndexMut and AsMut are safe as they don't support extending the vec. Both just give you multiple access to the vec elements.